### PR TITLE
feat: Add support for modifiable retention policies & enable deleting retention policy assignment

### PR DESCRIPTION
**Pull request by arjankowski**

### CODE REVIEW

File: src/commands/retention-policies/create.js [hunk 1]
```diff
@@ -46,6 +52,33 @@ RetentionPoliciesCreateCommand.flags = {
...
+	'retention-type': flags.string({
+		description: 'The type of retention for the new policy',
+		exclusive: [
+			'modifiable',
+			'non-modifiable'
+		],
+		options: [
+			'modifiable',
+			'non_modifiable'
+		]
```
Comment (mwwoda) [inline src/commands/retention-policies/create.js:64]: Can you elaborate a bit on the reason for including the option to set modifiable as string? I thought this was done to allow new values without changing the code in the future (as it cannot be done with boolean flags), but I see that we are still limiting the possible options.

File: src/commands/retention-policies/update.js [hunk 2]
```diff
@@ -20,6 +20,9 @@ class RetentionPoliciesUpdateCommand extends BoxCommand {
 		if (flags['retention-length']) {
 			updates.retention_length = parseInt(flags['retention-length'], 10);
 		}
+		if (flags['non-modifiable']) {
+			updates.retention_type = this.client.retentionPolicies.retentionTypes.NON_MODIFIABLE;
+		}
```
Comment (mwwoda) [inline src/commands/retention-policies/update.js:25]: Just to clairify - we cannot update the policy to set it to `modifiable` right? We can only set it `non-modifiable` once it was created?

File: src/commands/retention-policies/create.js (hunk 1 above)
Comment (arjankowski) [reply src/commands/retention-policies/create.js:64]: I added this to be compatible with update method, when we can only update type to `non-modifiable`

File: src/commands/retention-policies/update.js (hunk 2 above)
Comment (arjankowski) [reply src/commands/retention-policies/update.js:25]: You can create retention policy with either `modifiable` or `non-modifiable` type,  but you cannot change a policy to `modifiable` if it has been created `non-modifiable`.

**Review by mwwoda — APPROVED** [review]

**Pull request merged**